### DISCUSSION TITLE
Is it possible to create a single file of server to ship with new version of our app

### DISCUSSION THREAD

**Discussion by iampopal** [serverpod/serverpod#1280] [General]
We have a Store Management System with which each store owner can create a new database, and open that database by clicking on it and we are currently using SQLite.

Is it possible to have a UI for each store to create a new Database with their own server password and then open that database by clicking on it? 

We need to ship the servers with our app, as users click on their database file there server shall start on user's computer or mobile. 

Is it possible with Serverpod to have a single server file and start a new database with that server in windows, macos and android/ios without docker?

**Comment by Isakdl** [comment]
What do you want to achieve here? Too me it sounds like what you are doing is an offline app where you just want persistence of data, am I missing something?

If that is the case I think Serverpod would not be the right tool for the job. Serverpod is intended as a backend to manage data off the users devices, and you should host the Serverpod instance in the cloud exposed to the internet.

For local persistence there are better more lightweight solutions that you can use, such as SQLite (like you already have?), or some middle-ware in your state management lib in your app.

**Comment by iampopal** [comment]
Thank you for your replay,

Currently we need persistence of multiple users offline,
Like a hospital management system which will have multiple doctors,
libertarians, and medicine.
We need it offline now because in our area there is no stable internet
connection everywhere.

But in future when internet get better here we are thinking to host our
users database in server and each hospital will have an online database.

Because of current low internet connection we need offline first and if a
user want full online system we host it in a container for them.

We have already tried sqlite in other of our projects and it worked well
but we are thinking about adding more features to each hospital management
system in future which may not supported by sqlite.

So is it possible to compile into single build and run it anywhere in
windows, mac and in future in container. Without doing extra setup by our
users. Like we do in sqlite.

**Reply by Isakdl** [reply]
Serverpod is not intended to run on the client machine, in theory I suppose you could, but you will need to install a postgresql instance along with the Serverpod binary.  But keep in mind you will lose all security features if you ship the server code to the client as the client now have full access to the code and data. I would highly discourage you from taking this approach as you would be using the wrong tool for the job. 

If you want to build an app that is offline ONLY, then you do not need a backend at all, a simple persistence layer client side should be enough.

If you actually need a backend to sync data between multiple clients with a shared database model, then you will need to host this somewhere. Either in the cloud like you normally would, or another potential solution is to host an instance in the local network, but this would mean you have to take care of the server hardware and configurations. 

Between these two I would rather opt for hosting in the cloud if possible, but building the clients to work offline and then sync the state when ever there is a working internet connection.

I hope this helps! :)

**Comment by iampopal** [comment]
Thank you clarification,

We need a local server to host our Database because of there are multiple
users saving data in system.

And we need it full-offline because mostly of our users not has
internet access they like to use it in their local server.

But we need a way so if our database get large or some of them need cloud
database hosting, we easily able to host it for them in cloud server too.

**Comment by iampopal** [comment]
Offline first and sync is also good idea,

But we are fearing as there will be one user saving data and another user
getting it in realtime so we believe that the need of live system to host
it in local server is more important.

Like:
Appointment admin: save a new appointment for doctor A

Doctor A: when the patient come to Doctor A he shall see in list of
participants a new doctor.

So because of above needs:
We need to ship postgressql with our dart server with final version of our
system

So each user will create a database for them locally,
Can you please show me steps or simple of app which build a dart server
executable with postgressql?
